src/main/java/com/minecolonies/coremod/commands/colonycommands/ColonyTeleportCommand.java

```diff
@@ -56,14 +73,73 @@ public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final En
         return colony.getPermissions().hasPermission(player, Action.TELEPORT_TO_COLONY);
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        if (null == colony)
+        {
+            final EntityPlayer player = actionMenu.getPlayerForArgument("player");
+            if (player != null)
+            {
+                IColony iColony = ColonyManager.getIColonyByOwner(server.getEntityWorld(), player);
+                if (null == iColony)
+                {
+                    if (sender instanceof EntityPlayer)
+                    {
+                        final Entity senderEntity = sender.getCommandSenderEntity();
+                        if (senderEntity != null)
+                        {
+                            final UUID mayorID = senderEntity.getUniqueID();
+                            if (iColony == null)
+                            {
+                                iColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID);
+                            }
+                        }
+                    }
+                }
+
+                if (null != iColony)
+                {
+                    colony = ColonyManager.getColony(iColony.getID());
+                }
+            }
+        }
+
+        executeShared(server, sender, colony);
+    }
+
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        Colony colony = null;
+        //see if player is allowed to use in the configs
+        if (args.length == 1)
+        {
+            try
+            {
+                final int colonyId = Integer.valueOf(args[0]);
+                colony = ColonyManager.getColony(colonyId);
+            }
+            catch (final NumberFormatException e)
+            {
+            }
+        }
+
+        if (null == colony)
+        {
+            sender.sendMessage(new TextComponentString("You are not allowed to do this"));
+        }
+
+        executeShared(server, sender, colony);
+    }
+
+    private void executeShared(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final Colony colony)
     {
         //see if player is allowed to use in the configs
-        if (sender instanceof EntityPlayer && args.length == 1)
+        if (sender instanceof EntityPlayer)
         {
-            final int colonyID = getIthArgument(args, 0, -1);
-            if (colonyID != -1 && canPlayerUseCommand((EntityPlayer) sender, COLONYTP, colonyID))
+            if (canPlayerUseCommand((EntityPlayer) sender, COLONYTP, colony.getID()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "colony" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)